catkin_ws/src/planner/src/substates/breadth_first_search.py

```diff
@@ -16,37 +16,25 @@ def __init__(self, expansionAmt, control, mapping, target_class, min_objects, ti
         self.expansionAmt = expansionAmt
         self.target_class = target_class
         self.min_objects = min_objects
+        self.search_depth = search_depth
 
     def doBreadthFirstSearch(self):
-        global moving
-        moving = False                
-            
-        movement = [0,1,0]
-
+        movement = [0,self.expansionAmt,0]
         while not rospy.is_shutdown():
             #move left
             print("Moving by {}.".format(movement))
-            moving = True
-            # [TODO] FIX
-            self.control.moveDeltaLocal(movement, callback=movementComplete, face_destination=True)
-            #check for object detected while moving
-            while moving and not rospy.is_shutdown():
-                if self.detectedObject: return # stop grid search when object found
+            self.control.moveDeltaLocal(movement, face_destination=True)
+            if self.detectedObject: return # stop grid search when object found
             #move left by the same amount again
```

Comment:
why move again?

Reply:
moving for breadth first search its like going in a square that gets bigger